src/audio/mux/mux.c

```diff
@@ -719,12 +731,19 @@ static int mux_prepare(struct comp_dev *dev)
 								source_list);
 			struct comp_buffer __sparse_cache *sink_c = buffer_acquire(sink);
 			int state = -1;
+			int direction = -1;
 
-			if (sink_c->sink)
+			if (sink_c->sink) {
 				state = sink_c->sink->state;
+				direction = sink_c->sink->direction;
+			}
 
 			buffer_release(sink_c);
 
+			/* the state is independent of loopback-typed sinks */
+			if (direction == SOF_IPC_STREAM_PLAYBACK)
+				continue;
```

Comment:
And in this case you only deal loopbacks with the capture path

Reply:
@johnylin76 any comment here ? am I wrong in my assumptions here ?